### PR TITLE
SCAL-151567 : updated resolveRequestBody to make all optional boolean…

### PR DESCRIPTION
**Pull request by sastaachar**
[SCAL-151567](https://thoughtspot.atlassian.net/browse/SCAL-151567), added extra attribute nullable and setting it to true, when a boolean param is not required.

### CODE REVIEW

File: test/schema.graphql [hunk 1]
```diff
@@ -380,6 +380,10 @@ type Query {
         COMPACT: The response includes only the value of the columns.
         """
         formatType: FormatData = COMPACT
+        """ 
+        This is an optional boolean type and is nullable
+        """
+        boolType: Boolean
```
Comment (abdullasulaiman) [inline test/schema.graphql:386]: remove this unwanted one for now or add a better example for the same to validate

File: src/operations.ts [hunk 2]
```diff
@@ -73,6 +80,7 @@ function resolveRequestBody(args: any[]) {
             default: arg.defaultValue,
             ...resolveFieldType(arg.type),
             deprecated: !!arg.deprecationReason,
+            nullable: isNullableType(arg),
```
Comment (abdullasulaiman) [inline src/operations.ts:83]: This property should be added as optional something like below
...(isNullableType(arg) && { nullable: isNullableType(arg) }),

**Review by abdullasulaiman — CHANGES_REQUESTED** [review]

File: test/schema.graphql (hunk 1 above)
Comment (sastaachar) [reply test/schema.graphql:386]: added this to validate the changes in the tests

File: src/operations.ts (hunk 2 above)
Comment (sastaachar) [reply src/operations.ts:83]: yes will update this

**Review by abdullasulaiman — APPROVED** [review]

**Review by ashubham — APPROVED** [review]

**Pull request merged**